name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/DABPDFExtractor.java

```diff
@@ -60,47 +60,55 @@ private void addBuyTransaction()
                         .section("isin", "name", "currency") //
                         .find("Gattungsbezeichnung ISIN") //
                         .match("^(?<name>.*) (?<isin>[^ ]*)$")
-                        .match("STK [\\d.]+(,\\d+)? (?<currency>\\w{3}+) ([\\d.]+,\\d+)$")
+                        .match("STK [\\d\\.]+(,\\d+)? (?<currency>\\w{3}) ([\\d\\.]+,\\d+)$")
                         .assign((t, v) -> t.setSecurity(getOrCreateSecurity(v)))
 
                         .section("shares") //
                         .find("Nominal Kurs") //
-                        .match("^STK (?<shares>[\\d.]+(,\\d+)?) (\\w{3}+) ([\\d.]+,\\d+)$")
+                        .match("^STK (?<shares>[\\d\\.]+(,\\d+)?) (\\w{3}) ([\\d\\.]+,\\d+)$")
                         .assign((t, v) -> t.setShares(asShares(v.get("shares"))))
 
                         .section("amount", "currency") //
                         .optional() //
-                        .find("Wert Konto-Nr. Betrag zu Ihren Lasten")
-                        .match("^(\\d+.\\d+.\\d{4}+) ([0-9]*) (?<currency>\\w{3}+) (?<amount>[\\d.]+,\\d+)$")
+                        .find("Wert Konto-Nr\\. Betrag zu Ihren Lasten")
+                        .match("^(\\d+\\.\\d+\\.\\d{4}) ([0-9]*) (?<currency>\\w{3}) (?<amount>[\\d\\.]+,\\d+)$")
```

Comment:
It’s different here with the first few dots, because here they are used outside of character classes. An unescaped dot there matches any character, so the old date pattern would match `27X12Y2020`, too. So, changing it here is correct (but not in the “amount” part, where it is again in a character class).